index.js

```diff
@@ -92,6 +97,38 @@ function isFormMethodDialog(el) {
   return el.getAttribute('method').toLowerCase() === 'dialog';
 }
 
+/**
+ * @param {!DocumentFragment|!Element} hostElement
+ * @return {?Element} 
+ */
+function findFocusableElementInShadowDom(hostElement) {
+  // Note that this is 'any focusable area'. This list is probably not exhaustive, but the
+  // alternative involves stepping through and trying to focus everything.
+  var opts = ['button', 'input', 'keygen', 'select', 'textarea'];
+  var query = opts.map(function(el) {
+    return el + ':not([disabled])';
+  });
+  // TODO(samthor): tabindex values that are not numeric are not focusable.
+  query.push('[tabindex]:not([disabled]):not([tabindex=""])');  // tabindex != "", not disabled
+  target = hostElement.querySelector(query.join(', '));
+
+  if (!target) {
```

Comment:
I, uh... am not a huge fan of this, because it amounts to walking the entire tree.

I'm going to ask folks on my team about better ways to do this, if any.

Reply:
There is no deep selector that I know of into Shadow DOM. You only walk the whole tree _until_ you find a focusable item.

I'm open to an opt-in option to trigger this behavior.

Reply:
Can we make sure this checks for existence of Shadow DOM? Otherwise we're iterating over everything and checking for `shadowRoot` even though maybe the feature doesn't exist.

Reply:
Fixed.